### ISSUE TITLE
Code Table Request - New Taxonomy Source: IDNR List

### ISSUE THREAD

**Issue by Jegelewicz** [ArctosDB/arctos#6768]
### Goal
_Describe what you're trying to accomplish. This is the only necessary step to start this process. The Committee is available to assist with all other steps. Please clearly indicate any uncertainty or desired guidance if you proceed beyond this step._

Track threatened/endangered species in the state of Illinois

### Context
_Describe why this new value is necessary and existing values are not._

This source will be updated manually any time a new IDNR threatened/endangered list is released. The single source will make that easy. It has been tested and found to be a good solution - see #5444 

### Table
_Code Tables are http://arctos.database.museum/info/ctDocumentation.cfm. Link to the specific table or value. This may involve multiple tables and will control datatype for Attributes. OtherID requests require BaseURL (and example) or explanation. Please ask for assistance if unsure._

https://arctos.database.museum/info/ctDocumentation.cfm?table=cttaxonomy_source

### Proposed Value
_Proposed new value. This should be clear and compatible with similar values in the relevant table and across Arctos._

IDNR List

### Proposed Definition
_Clear, complete, non-collection-type-specific **functional** definition of the value. Avoid discipline-specific terminology if possible, include parenthetically if unavoidable._

A source that records the endangered/threatened status of species in the state of Illinois. This source is expected to only include non-classification terms and must be manually updated each time Illinois Department of Natural Resources releases a new list. This source is managed by Chicago Academy of Sciences (CHAS).

### Collection type
_Some code tables contain collection-type-specific values. ``collection_cde`` may be found from https://arctos.database.museum/home.cfm_

N/A

#### Attribute data type
_If the request is for an attribute, what values will be allowed?
free-text, categorical, or number+units depending upon the attribute (TBA)_

N/A

#### Attribute controlled values
_If the values are categorical (to be controlled by a code table), add a link to the appropriate code table. If a new table or set of values is needed, please elaborate._

N/A

#### Attribute units
_if numerical values should be accompanied by units, provide a link to the appropriate units table._

N/A

#### Part preservation attribute affect on "tissueness"
_if a new part preservation is requested, please add the affect it would have on "tissueness": No Influence, Allows, or Denies_

N/A

### Priority
_Please describe the urgency and/or choose a priority-label to the right. You should expect a response within two working days, and may utilize [Arctos Contacts](https://arctosdb.org/contacts/) if you feel response is lacking._

### Example Data
_Requests with clarifying sample data are generally much easier to understand and prioritize. Please attach or link to any representative data, in any form or format, which might help clarify the request._

See #5444 

### Available for Public View
_Most data are by default publicly available. Describe any necessary access restrictions._

Yes

### Helpful Actions

- [x] Add the issue to the [Code Table Management Project](https://github.com/ArctosDB/arctos/projects/13#card-31628184).

- [x] Please reach out to anyone who might be affected by this change. Leave a comment or add this to the Committee agenda if you believe more focused conversation is necessary.

@ArctosDB/arctos-code-table-administrators 

## Approval
_All of the following must be checked before this may proceed._

_The [How-To Document](https://handbook.arctosdb.org/how_to/How-To-Manage-Code-Table-Requests.html) should be followed. Pay particular attention to terminology (with emphasis on consistency) and documentation (with emphasis on functionality). **No person should act in multiple roles**; the submitter cannot also serve as a Code Table Administrator, for example._

- [x] Code Table Administrator[1] - check and initial, comment, or thumbs-up to indicate that the request complies with the how-to documentation and has your approval
- [x] Code Table Administrator[2] - check and initial, comment, or thumbs-up to indicate that the request complies with the how-to documentation and has your approval
- [x] DBA - The request is functionally acceptable. The term is not a functional duplicate, and is compatible with existing data and code.
- [x] DBA - Appropriate code or handlers are in place as necessary. (ID_References, Media Relationships, Encumbrances, etc. require particular attention)

## Rejection
_If you believe this request should not proceed, explain why here. Suggest any changes that would make the change acceptable, alternate (usually existing) paths to the same goals, etc._

1. _Can a suitable solution be found here? If not, proceed to (2)_
2. _Can a suitable solution be found by Code Table Committee discussion? If not, proceed to (3)_
3. _Take the discussion to a monthly Arctos Working Group meeting for final resolution._

## Implementation
_Once all of the Approval Checklist is appropriately checked and there are no Rejection comments, or in special circumstances by decree of the Arctos Working Group, the change may be made._

- [x] Review everything one last time.  Ensure the How-To has been followed. Ensure all checks have been made by appropriate personnel.

- [x] Add or revise the code table term/definition as described above. Ensure the URL of this Issue is included in the definition. URLs should be included *_as text_*, separated by spaced pipes. Do not include HTML in definitions.

_Close this Issue._

_**DO NOT** modify Arctos Authorities in any way before all points in this Issue have been fully addressed; data loss may result._

## Special Exemptions

_In very specific cases and by prior approval of The Committee, the approval process may be skipped, and implementation requirements may be slightly altered. Please note here if you are proceeding under one of these use cases._

1. _Adding an existing term to additional collection types may proceed immediately and without discussion, but doing so may also subject users to future cleanup efforts. If time allows, please review the term and definition as part of this step._
2. _The Committee may grant special access on particular tables to particular users. This should be exercised with great caution only after several smooth test cases, and generally limited to "taxonomy-like" data such as International Commission on Stratigraphy terminology._

**Comment by Nicole-Ridgwell-NMMNHS** [comment]
Looking through the other issue, this seems like a good solution. Checking a box.

**Comment by Jegelewicz** [comment]
@ArctosDB/arctos-code-table-administrators anyone besides Nicole willing to check a box?

**Comment by Jegelewicz** [comment]
@ArctosDB/arctos-code-table-administrators can we PLEASE get a box checked so I can move forward with this? If there is some reason NOT to add this, please speak up.

**Comment by campmlc** [comment]
I support this - I'll check it off if no objections.

**Comment by campmlc** [comment]
Box checked.

**Comment by Jegelewicz** [comment]
The term has been added to the code table.

The following taxa were added to Arctos to accommodate the list

SCIENTIFIC_NAME
--
Carex cryptolepsis
Gratiola quartermaniae
Lycaeides melissa samuelis
Prostoia ozarkensis
Pygmarrhopalites madonnensis
Symphoricarpos albus var. albus

The following taxa are mispelled in the IDNR list and the classification will be associated with the Arctos taxon:

SCIENTIFIC_NAME | STATUS | SUGGESTED_SCI_NAME
-- | -- | --
Fundulus diaphanus subsp. menona | spelling | Fundulus diaphanus menona
Hyloteliphium telephioides | spelling | Hylotelephium telephioides
Lathryus ochroleucus | spelling | Lathyrus ochroleucus

**Comment by Jegelewicz** [comment]
Waiting for new source to show in the code table, then will upload classifications.

**Comment by Jegelewicz** [comment]
@dustymc I was trying to use the new metadata in https://arctos.database.museum/info/ctDocumentation.cfm?table=cttaxonomy_source

And I added "Arctos Classification Bulkloader" to the new IDNR List source - which gets me this in the [bulkload classification tool](https://arctos.database.museum/tools/BulkloadClassification.cfm):

![image](https://github.com/ArctosDB/arctos/assets/5725767/ce295f4f-d851-4597-bc03-7bf7634d8476)

Please 'splain me what I did wrong!

**Comment by Jegelewicz** [comment]
OOOH - maybe because I wasn't an authorized editor? I changed it - let's see if that works....

**Comment by dustymc** [comment]
>I changed it 

Thx, was a headscratcher....

The check is cached, give it an hour.

**Comment by Jegelewicz** [comment]
OK they loaded, but trying to search by the source gives me a weird error (and maybe I am not doing it correctly)

## Search parameters

collection: CHAS:Mamm,CHAS:Bird
taxon term: %
type or rank: idnr_status
source: IDNR List

## Error:
![image](https://github.com/ArctosDB/arctos/assets/5725767/ab34b589-833b-45d1-92b1-e6efa01306b1)

Trying to get all of the records for those collections that are identified as something with a term in the IDNR List source.

**Comment by Jegelewicz** [comment]
Above is sorta equivalent to what we did in test - https://github.com/ArctosDB/arctos/issues/5444#issuecomment-1432190507

**Comment by dustymc** [comment]
https://arctos.database.museum/search.cfm?guid_prefix=CHAS%3ABird%2CCHAS%3AMamm&tax_rnk_1=idnr_status&tax_src_1=IDNR%20List

¯\\_(ツ)_/¯

**Comment by Jegelewicz** [comment]
Question - does the collection need to using the source for this to work? I just tried again, not logged in using Chrome and I get 

![image](https://github.com/ArctosDB/arctos/assets/5725767/7f55a818-f060-4da6-acf3-a61b8c2174ea)

**Comment by dustymc** [comment]
And https://github.com/ArctosDB/arctos/issues/6311 but this still seems to confuse people, maybe we need a video or some sort or something??

**Comment by dustymc** [comment]
> does the collection need to using the source

Nope, and wouldn't be much reason to pull globalnames if that was necessary.

<img width="1683" alt="Screenshot 2023-11-03 at 12 25 21" src="https://github.com/ArctosDB/arctos/assets/5720791/ee00c3a5-bc3f-4fa2-a3f9-d21f193a8fd6">

**Comment by Jegelewicz** [comment]
But with your shareable link it works...

![image](https://github.com/ArctosDB/arctos/assets/5725767/4bc1c265-1724-492d-983f-6a47d87c9bac)

However in your shared link the term looks like this ( a crazy little underscore) - why not a %

![image](https://github.com/ArctosDB/arctos/assets/5725767/022283a3-b456-4754-9ba3-f6c07cc16dec)

**Comment by Jegelewicz** [comment]
This is really nice if we can un-confuse the % vs _ thing.  CHAS can find all of their state endangered stuff:

https://arctos.database.museum/search.cfm?guid_prefix=CHAS%3ABird%2CCHAS%3AMamm&tax_trm_1=_&tax_rnk_1=idnr_status&tax_src_1=IDNR%20List

![image](https://github.com/ArctosDB/arctos/assets/5725767/5052255d-fa36-4fcb-8580-2d5d6bf93610)

**Comment by Jegelewicz** [comment]
I was just following instructions...

![image](https://github.com/ArctosDB/arctos/assets/5725767/70011935-c718-4c9b-ad59-3d4847d3bfb0)

**Comment by dustymc** [comment]
>why not a %

File an issue - I think it should probably be that source OR term OR value OR any/all do stuff, but IDK if that's affordable or desirable or ?? 

>instructions

Include that too, I can change the placeholder, you can change the help text, that's accurate except when it's prefixing nothing....

**Comment by Jegelewicz** [comment]
New issue - https://github.com/ArctosDB/arctos/issues/6923

Added for demo at next AWG meeting.

@wellerjes @droberts49 this is now functional for you in production!

https://arctos.database.museum/search.cfm?guid_prefix=CHAS%3ABird%2CCHAS%3AMamm&tax_trm_1=_&tax_rnk_1=idnr_status&tax_src_1=IDNR%20List

Closing.

**Comment by wellerjes** [comment]
Thank you!
